src/modules/claims/claims.service.ts

```diff
@@ -269,20 +269,50 @@ export class ClaimsService {
   async registerOnchain(
     claim: Pick<Claim, 'token' | 'subjectAgreement' | 'onChainProof' | 'acceptedBy'>
   ) {
-    if (!readyToBeRegisteredOnchain(claim)) {
+    if (!readyToBeRegisteredOnchain(claim) || !claim.token) {
```

Comment:
I don't think this is necessary as token already required by `readyToBeRegisteredOnchain`

Reply:
sorted